### PR TITLE
test(shjs): add tests for shjs

### LINKED ISSUE #280: Add tests for the shjs utility
It might be useful to add some tests for the `shjs` utility itself. This would include making sure it respects stdout/stderr and exit codes. This is less of a priority now, but would be more useful if #268 (stderr) and #252 (exit codes) are merged.

This would involve creating shelljs scripts and executing them with the `shjs` command, and making sure results are as expected. I don't think we need a large number of test cases, but we should have a few just to make sure we don't get regressions in the utility.

### PR DESCRIPTION
**Pull request by ariporad**
Fixes  #280.

### CODE REVIEW

**Comment by ariporad** [comment]
@nfischer: ready for review.

File: test/resources/shjs/exit-codes.js [hunk 1]
```diff
@@ -0,0 +1 @@
+process.exit(42);
```
Comment (nfischer) [inline test/resources/shjs/exit-codes.js]: Can we add a test with return code `0`?

Comment (ariporad) [inline test/resources/shjs/exit-codes.js]: Sure

File: package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
   },
   "dependencies": {},
   "devDependencies": {
+    "coffee-script": "^1.10.0",
```
Comment (nfischer) [inline package.json]: This dependency seems reasonable to me :+1:

File: test/shjs.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+function runScript(name) {
+  return shell.exec(path.resolve(__dirname, '../bin/shjs') +
+                    ' ' +
+                    path.resolve(__dirname, 'resources', 'shjs', name), { silent: true });
+}
+
+assert(runScript('exit-codes.js'), 42, 'exit code works');
+
+var stdioRet = runScript('stdout-stderr.js');
+assert(stdioRet.stdout, 'stdout: OK!\n', 'stdout works');
```
Comment (nfischer) [inline test/shjs.js]: s/assert/assert.equal

Also, I don't think it takes a third parameter

File: test/shjs.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+assert(stdioRet.stderr, 'stderr: OK!\n', 'stderr works');
+
+assert(runScript('coffeescript.coffee').stdout, 'CoffeeScript: OK!\n');
```
Comment (nfischer) [inline test/shjs.js]: assert.equal, like above. Also, perhaps we could check its error code to make sure it's 0?

File: test/resources/shjs/exit-codes.js (hunk 1 above)
Comment (nfischer) [inline test/resources/shjs/exit-codes.js]: (or just see my comment below)

**Comment by nfischer** [comment]
Could we add a test to make sure it correctly infers the file extension? Perhaps `shjs path/to/resource/file` for a file named `path/to/resource/file.js`?

**Comment by nfischer** [comment]
Other than that, LGTM

File: test/shjs.js (hunk 3 above)
Comment (ariporad) [inline test/shjs.js]: Yes it does. (All assert methods do), it's a error message if the assert fails.

**Comment by ariporad** [comment]
@nfischer: Should have fixed all those, feel free to re-review/merge.

**Comment by nfischer** [comment]
Awesome!

**Pull request merged**